### ISSUE TITLE
[Bug] Mamba-3 decay always produces `A==+A_floor`

### ISSUE THREAD

**Issue by netanel-haber** [fla-org/flash-linear-attention#1013]
Mamba-3 negates `A` after clamping the positive softplus result:

```python
A = -F.softplus(dd_A.to(torch.float32)).clamp(max=-self.A_floor)
```

Because `softplus(dd_A) >= 0` and `A_floor > 0`, this always produces `A == +A_floor` and gives `dd_A` zero gradient.

```python
import torch
import torch.nn.functional as F

dd_A = torch.tensor([0.0])
A_floor = 0.1

actual = -F.softplus(dd_A).clamp(max=-A_floor)
expected = (-F.softplus(dd_A)).clamp(max=-A_floor)

print(actual)    # tensor([0.1000])
print(expected)  # tensor([-0.6931])
```

`A` should be negated before clamping:

```python
A = -F.softplus(dd_A.to(torch.float32))
A = A.clamp(max=-self.A_floor)
```

Fixed by #1012.
